sympy/stats/tests/test_symbolic_probability.py

```diff
@@ -22,11 +22,21 @@ def test_literal_probability():
     assert Expectation(X).rewrite(Integral).doit() == expectation(X)
     assert Expectation(X**2).evaluate_integral() == expectation(X**2)
     assert Expectation(x*X).args == (x*X,)
-    assert Expectation(x*X).doit() == x*Expectation(X)
-    assert Expectation(2*X + 3*Y + z*X*Y).doit() == 2*Expectation(X) + 3*Expectation(Y) + z*Expectation(X*Y)
+    assert Expectation(x*X).expand() == x*Expectation(X)
+    assert Expectation(2*X + 3*Y + z*X*Y).expand() == 2*Expectation(X) + 3*Expectation(Y) + z*Expectation(X*Y)
```

Comment:
What about keeping both `.doit()` and `.expand()` in the tests?

Reply:
Do you mean using them simultaneously?

Reply:
I mean... what happens if you call `.doit()` on this expression? Should be add an assertion?

Reply:
It falis for expression having CRV in trigonometric functions due to complex integration
```python
>>> X = Normal('X', 2, 3)
>>> Y = Normal('Y', 3, 4)
>>> Z = Poisson('Z', 4)
>>> W = Poisson('W', 3)
>>> x = symbols('x', real=True)
>>> y, w, z = symbols('y, w, z')
>>> Expectation(2*X + 3*Y + z*X*Y).expand()
z*Expectation(X*Y) + 2*Expectation(X) + 3*Expectation(Y)
>>> Expectation(2*X + 3*Y + z*X*Y).doit()
6*z + 13
>>> Expectation(sin(X)).doit() # doesn't work for CRV
Traceback (most recent call last):
  ...
TypeError: Invalid comparison of non-real I*im(exp_polar(-I*pi/2))*erf(3*sqrt(2)*I*exp_polar(0)*im(exp_polar(
I*pi/2))/2)
>>> X = Die('X')
>>> Expectation(sin(X)).doit()   # works fine with FRV
sin(5)/6 + sin(4)/6 + sin(6)/6 + sin(3)/6 + sin(1)/6 + sin(2)/6
>>> Expectation(sin(Z)).doit()  #  works with DRV but doesn't evaluate
Sum(4**Z*exp(-4)*sin(Z)/factorial(Z), (Z, 0, oo))

```

`>>> Expectation(sin(X)).doit() # doesn't work for CRV` Should I add this as XFAIL?

Reply:
The above fails in `master` as well. A new issue should be opened for fixing this. Do not add this as test for now.